controller/login.go

```diff
@@ -112,6 +115,70 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
 	return c.auth.Perform(ctx, oauth, brokerEndpoint, entitlementEndpoint, profileEndpoint, whitelist, c.configuration.GetAuthNotApprovedRedirect())
 }
 
+// getEntitlementResourceRequestPayload creates the object which would have the information about which spaces/resources
+// the entitlements' info would need to be fetched for.
+
+func (c *LoginController) getEntitlementResourceRequestPayload(ctx context.Context, token *string) (*auth.EntitlementResource, error) {
+	loggedInIdentityID, err := c.tokenManager.Extract(*token)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "unable to get ID from access token")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get ID from access token"))
+	}
+
+	// get the user object as well for this identity
+	queryResult, err := c.identityRepository.Query(account.IdentityFilterByID(loggedInIdentityID.ID), account.IdentityWithUser())
+	if err != nil || len(queryResult) == 0 {
+		log.Error(ctx, map[string]interface{}{
+			"err":         err,
+			"identity_id": *loggedInIdentityID,
+		}, "unable to query Identity")
+		return nil, errors.NewInternalError(ctx, errs.Wrap(err, "unable to query Identity"))
+	}
+	loggedInIdentity := queryResult[0]
+	contextInfoLoggedInIdentity := loggedInIdentity.User.ContextInformation
+	_, recentSpacesPresent := contextInfoLoggedInIdentity["recentSpaces"]
+	if contextInfoLoggedInIdentity == nil || !recentSpacesPresent {
+		log.Warn(ctx, map[string]interface{}{
+			"identity_id": *loggedInIdentityID,
+		}, "unable to find recentSpaces in ContextInformation")
+		return nil, nil
+	}
+
+	var spacesToGetEntitlementsFor []auth.ResourceSet
+	for _, v := range contextInfoLoggedInIdentity["recentSpaces"].([]interface{}) {
+		if len(spacesToGetEntitlementsFor) == maxRecentSpacesForRPT {
+			log.Info(ctx, map[string]interface{}{
+				"identity_id":               *loggedInIdentityID,
+				"max_recent_spaces_for_rpt": maxRecentSpacesForRPT,
```

Comment:
`time="2017-07-18T12:19:33+05:30" level=info msg="more than the allowed maximum number of recent spaces found" identity_id={{0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC <nil>} ae68a343-c866-430c-b6ce-a36f0b38d8e5 testuser false  <nil> {00000000-0000-0000-0000-000000000000 false} {{0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC <nil>} 00000000-0000-0000-0000-000000000000       [] map[]}} max_recent_spaces_for_rpt=10 pkg=controller req_id= total_number_of_recent_spaces=12`

done in 368b59ed88a0f02362d9e1981be4df111c5b399f